cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/metric/ContainerMetricUtils.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.metricsreporter.metric;
+
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+
+public class ContainerMetricUtils {
+
+    public double getCpuPeriod() {
+        return Double.parseDouble(readFile(CgroupFiles.PERIOD_PATH.getValue()));
+    }
+
+    public double getCpuQuota() {
+        return Double.parseDouble(readFile(CgroupFiles.QUOTA_PATH.getValue()));
+    }
+
+    /**
+     * Gets the the number of logical cores available to the node.
+     *
+     * We can get this value while running in a container by using the "nproc" command.
+     * Using other methods like OperatingSystemMXBean.getAvailableProcessors() and
+     * Runtime.getRuntime().availableProcessors() would require disabling container
+     * support (-XX:-UseContainerSupport) since these methods are aware of container
+     * boundaries
+     *
+     * @return Number of logical processors on node
+     */
+    public int getAvailableProcessors() {
+        int proc = 1;
+        try {
+            InputStream in = Runtime.getRuntime().exec("nproc").getInputStream();
+            proc = Integer.parseInt(readStringValue(in));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return proc;
+    }
+
+    private static String readFile(String path) {
+        String s = null;
+        try {
+            s = readStringValue(new FileInputStream(path));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return s;
+    }
+
+    private static String readStringValue(InputStream in) {
+        try {
+            BufferedReader br = new BufferedReader(new InputStreamReader(in, StandardCharsets.UTF_8));
+            String line;
+            while ((line = br.readLine()) != null) {
```

Comment:
> This function seems to just return the first line from the given input stream. If the stream is empty, it returns null. Can we drop the loop and rename the function to reflect what it does?

You are right, we don't need the `while` statement here,  I updated the function to just read the first line of the stream and renamed the function.

> -- also is it ever acceptable for this function to return null? Should we handle the null case either at this function or in its returned context?

No it is not, we should definitely handle the null case! Just added a null check in case the InputStream is empty/null.

Let me know what you think!